garde_derive/src/syntax.rs

```diff
@@ -354,27 +365,10 @@ impl Parse for model::RawLength {
                 }
             };
             match arg {
-                RawLengthArgument::Min(span, v) => {
-                    if min.is_some() {
-                        error.maybe_fold(syn::Error::new(span, "duplicate argument"));
-                        continue;
-                    }
-                    min = Some(v);
-                }
-                RawLengthArgument::Max(span, v) => {
-                    if max.is_some() {
-                        error.maybe_fold(syn::Error::new(span, "duplicate argument"));
-                        continue;
-                    }
-                    max = Some(v);
-                }
-                RawLengthArgument::Mode(span, v) => {
-                    if mode.is_some() {
-                        error.maybe_fold(syn::Error::new(span, "duplicate argument"));
-                        continue;
-                    }
-                    mode = Some(v);
-                }
+                RawLengthArgument::Min(span, v) => update_opt!(min, span, v),
```

Comment:
Please revert this back to not use a macro.

I personally use macros for implementing a bunch of boilerplate code on some list of things that is subject to change, or to DRY a pattern that is the exact same everywhere and there isn't a neat way to extract it without a macro.
The here code is duplicated 4 times, and when accounting for the added macro, the entire diff is only a few lines of code. It's also unlikely to change much more any time soon.